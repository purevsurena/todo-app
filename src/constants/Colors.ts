/**
 * Below are the colors that are used in the app. The colors are defined in the light and dark mode.
 * There are many other ways to style your app. For example, [Nativewind](https://www.nativewind.dev/), [Tamagui](https://tamagui.dev/), [unistyles](https://reactnativeunistyles.vercel.app), etc.
 */

const tintColorLight = "#0a7ea4";
const tintColorDark = "#fff";

export const Colors = {
  light: {
    text: "#11181C",
    background: "#fff",
    tint: tintColorLight,
    icon: "#687076",
    tabIconDefault: "#687076",
    tabIconSelected: tintColorLight,
  },
  dark: {
    text: "#ECEDEE",
    background: "#151718",
    tint: tintColorDark,
    icon: "#9BA1A6",
    tabIconDefault: "#9BA1A6",
    tabIconSelected: tintColorDark,    
  },
  generic: {
    dark: "#000",
    background: '#2c2c2e',
    cardBG: '#1c1c1e',
    greyText: '#9E9EA5',
    blue: '#54a0ff',
    orange: '#ff5722',
    turquoise: '#00adb5',
    yellow: '#FFFF00',
  }
};
