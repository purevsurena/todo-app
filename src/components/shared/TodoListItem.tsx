import React, { useState } from "react";
import { StyleSheet, View } from "react-native";
import Flex from "@/components/layouts/Flex";
import { Colors } from "@/constants/Colors";
import Click from "@/components/layouts/Click";
import * as Animatable from "react-native-animatable";
import BouncyCheckbox from "react-native-bouncy-checkbox";
import { Swipeable } from "react-native-gesture-handler";
import Icon from "@expo/vector-icons/MaterialCommunityIcons";

interface TodoListItemProps {
  description: string;
  color: string;
  index: number;
  isChecked: boolean;
  onToggle: () => void;
  onDelete: () => void;
}

const TodoListItem: React.FC<TodoListItemProps> = ({
  description,
  color,
  index,
  isChecked: initialChecked,
  onToggle,
  onDelete,
}) => {
  const [isChecked, setIsChecked] = useState(initialChecked);

  const handleToggle = () => {
    const newCheckedState = !isChecked;
    setIsChecked(newCheckedState);
    onToggle();
  };

  const renderRightActions = () => (
    <View style={styles.deleteButton}>
      <Click onPress={onDelete}>
        <Icon name="delete" size={20} color="red" />
      </Click>
    </View>
  );

  return (
    <Swipeable renderRightActions={renderRightActions}>
      <Click onPress={handleToggle} style={styles.container}>
        <Animatable.View
          animation="slideInUp"
          duration={(index + 1) * 50}
          style={{
            padding: 12,
            borderRadius: 16,
            backgroundColor: Colors.dark.background,
          }}
        >
          <Flex flexDirection="row" justifyContent="space-between">
            <Flex flexDirection="row" alignItems="center">
              <BouncyCheckbox
                size={20}
                fillColor={color}
                unFillColor={Colors.generic.cardBG}
                text={description}
                textStyle={{
                  color: isChecked
                    ? Colors.generic.greyText
                    : Colors.light.background,
                }}
                iconImageStyle={{ tintColor: Colors.generic.dark }}
                innerIconStyle={{ borderWidth: 1.5 }}
                isChecked={isChecked}
                onPress={handleToggle}
                disabled
              />
            </Flex>
          </Flex>
        </Animatable.View>
      </Click>
    </Swipeable>
  );
};

export default TodoListItem;

const styles = StyleSheet.create({
  container: {
    paddingTop: 12,
    paddingHorizontal: 16,
  },
  deleteButton: {
    marginTop: 12,
    justifyContent: "center",
    alignItems: "center",
    width: 75,
    backgroundColor: Colors.generic.label,
    borderRadius: 16,
  },
});
