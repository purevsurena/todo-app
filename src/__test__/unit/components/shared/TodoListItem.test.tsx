import React from "react";
import { render } from "@testing-library/react-native";
import TodoListItem from "@/components/shared/TodoListItem";
import { Colors } from "@/constants/Colors";

describe("TodoListItem Component", () => {
  it("should apply correct description based on prop", () => {
    const { getByText } = render(
      <TodoListItem
        description="Test Task"
        color={Colors.generic.blue}
        index={0}
        isChecked={true}
        onToggle={() => {}}
      />,
    );

    const description = getByText("Test Task");
    expect(description).toBeTruthy();
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(
      <TodoListItem
        description="Test Task"
        color={Colors.generic.blue}
        index={0}
        isChecked={false}
        onToggle={() => {}}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
