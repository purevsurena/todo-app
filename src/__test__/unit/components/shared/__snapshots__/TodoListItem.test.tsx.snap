// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem Component matches the snapshot 1`] = `
<View
  collapsable={false}
  handlerTag={3}
  handlerType="PanGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "flexDirection": "row-reverse",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#3C3C3C",
          "borderRadius": 16,
          "justifyContent": "center",
          "marginTop": 12,
          "width": 75,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "red",
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ó°†´
      </Text>
    </View>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    handlerTag={4}
    handlerType="TapGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      {
        "transform": [
          {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#151718",
          "borderRadius": 16,
          "padding": 12,
          "transform": [
            {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "alignItems": undefined,
              "alignSelf": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
            undefined,
          ]
        }
        testID="flexComponent"
      >
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "alignItems": "center",
                "alignSelf": undefined,
                "flex": undefined,
                "flexDirection": "row",
                "justifyContent": undefined,
              },
              undefined,
            ]
          }
          testID="flexComponent"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#1c1c1e",
                  "borderRadius": 10,
                  "height": 20,
                  "justifyContent": "center",
                  "transform": [
                    {
                      "scale": 1,
                    },
                  ],
                  "width": 20,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#54a0ff",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "height": 20,
                      "justifyContent": "center",
                      "width": 20,
                    },
                    {
                      "borderWidth": 1.5,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "flex": 1,
                    "marginLeft": 16,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#757575",
                      "fontSize": 16,
                      "textDecorationLine": "none",
                    },
                    {
                      "color": "#fff",
                    },
                  ]
                }
              >
                Test Task
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
